arch/risc-v/src/bl602/bl602_netdev.c

```diff
@@ -2120,10 +2122,12 @@ void bl602_net_event(int evt, int val)
     case CODE_WIFI_ON_DISCONNECT:
       do
         {
-          struct bl602_net_driver_s *priv = &g_bl602_net[0];
+          /* struct bl602_net_driver_s *priv = &g_bl602_net[0]; */
+
           if (g_state.sta_connected == 1)
             {
-              netdev_carrier_off(&priv->net_dev);
+              /* netdev_carrier_off(&priv->net_dev); */
```

Comment:
@SunJun8: Why this change? It seems to have been introduced in https://github.com/apache/incubator-nuttx/commit/79d35067fb9b03f52e0d8e383748a2c306ebf451, but I don't understand why.

Reply:
This looks like a change introduced due to a requirement. There may be some issues that need to be resolved at the time.

This change is outdated.

Reply:
It was included in https://github.com/apache/incubator-nuttx/commit/d489392d08e550e828146395d4e945de9952e6e3 (which was merged) though. To me it makes more sense to _have_ the call to `netdev_carrier_off` in `CODE_WIFI_ON_DISCONNECT` (so that userspace can detect the disconnection), and to _not_ have the call to `wifi_mgmr_sta_autoconnect_disable` in `CODE_WIFI_ON_CONNECTED` (so that `CODE_WIFI_CMD_RECONNECT` can trigger correctly). Thoughts?

Reply:
yes... `netdev_carrier_off`should be called here and it is used like this in unmerged internal code.

Reply:
I will update this part of the code in due course